magento/devdocs#: Resolvers. Add a tip about input/output types (#4331)
magento/devdocs: guides/v2.3/graphql/develop/resolvers.md

`@resolver(class)` | The class of the resolver
`@doc(description)` | Describes the purpose of the mutation

{:.bs-callout .bs-callout-tip}

> It's a good practice to define a separate types for input/output data. It bring additional extension points, so every input/output type can be extended by adding additional fields to the definition.

Reviewer: ```suggestion
It is a good practice to define separate types for input and output data. This practice permits additional extension points, so every input and output type can be extended by adding additional fields to the definition.
```

Author: **fixed**

Thank you, @keharper 👍

`@resolver(class)` | The class of the resolver
`@doc(description)` | Describes the purpose of the mutation

> {:.bs-callout .bs-callout-tip}

Reviewer 2: I would extend it with a simple example. I.e

Wrong approach:
```graphql
type Mutation {
    mutationQueryName(param1: String, param2: Int, ...): MutationQueryOutput @resolver(class: "Magento\\<module_name>\\Model\\Resolver\\MutationResolverModel") @doc(description:"Mutation query description")
}
```

Correct approach:
```graphql
type Mutation {
    mutationQueryName(inputParam: InputParamsType): MutationQueryOutput @resolver(class: "Magento\\<module_name>\\Model\\Resolver\\MutationResolverModel") @doc(description:"Mutation query description")
}

type InputParamsType {
    param1: String
    param2: Int
}
```

Thank you.

Author: **added**

Thank you, @rogyar 👍

Reviewer: I like the examples. Thanks. 

I'm checking with the internal GraphQL team, since we have mutations that don't follow this advice.

`generateCustomerToken(email: String! password: String!): CustomerToken`

Author: This is a proposition regards to BEST PRACTICES ;)

Many thanks, @keharper !

Author: @keharper , did you have a chance to verify changes in this PR?

Thank you!
